porch/repository/pkg/git/package_tree.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package git
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/filemode"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"k8s.io/klog/v2"
+)
+
+// packageList holds a list of packages in the git repository
+type packageList struct {
+	// parent is the gitRepository of which this is part
+	parent *gitRepository
+
+	// commit is the commit at which we scanned for packages
+	commit *object.Commit
+
+	// packages holds the packages we found
+	packages map[string]*packageListEntry
+}
+
+// packageListEntry is a single package found in a git repository
+type packageListEntry struct {
+	// parent is the packageList of which we are part
+	parent *packageList
+
+	// Path is the relative path to the root of the package (directory containing the Kptfile)
+	Path string
```

Comment:
Is `Path` intentionally exported?

Reply:
Ah ... it was used at one stage, but you're right, it's no longer used.  I will unexport till we need it.